lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -213,11 +213,24 @@ def check_BinaryExpression(o)
     rvalue(o.right_expr)
   end
 
+  def resource_without_title?(o)
+    if o.instance_of?(Puppet::Pops::Model::BlockExpression)
+      statements = o.statements
+      statements.length == 2 && statements[0].instance_of?(Puppet::Pops::Model::QualifiedName) && statements[1].instance_of?(Puppet::Pops::Model::LiteralHash)
+    else
+      false
+    end
+  end
+
   def check_BlockExpression(o)
-    o.statements[0..-2].each do |statement|
-      if idem(statement)
-        acceptor.accept(Issues::IDEM_EXPRESSION_NOT_LAST, statement)
-        break # only flag the first
+    if resource_without_title?(o)
+      acceptor.accept(Issues::RESOURCE_WITHOUT_TITLE, o, :name => o.statements[0].value)
+    else
+      o.statements[0..-2].each do |statement|
+        if idem(statement)
+          acceptor.accept(Issues::IDEM_EXPRESSION_NOT_LAST, statement)
+          break # only flag the first
+        end
```

Comment:
the above changes looks strange - it assumes there is always a block with exactly a name and a hash - clearly the error can occur first, last, in the middle of a block - I am not sure that this impl covers all cases.

OTOH, it may be that the rewrite rules for function calls and resouces always produce exactly such a block as the one being found by `resource_without_title` - is that really the case?